tests/gtfs/test_validation.py

```diff
@@ -580,3 +602,233 @@ def test_summarise_routes_on_pass(self, gtfs_fixture):
             "Size of date_route_counts not as expected. "
             "Expected {expected_size}"
         )
+
+    def test__plot_summary_defences(self, gtfs_fixture):
+        """Test defences for _plot_summary()."""
+        current_fixture = gtfs_fixture
+        current_fixture.summarise_routes()
+
+        # test parameters that are yet to be tested
+        with pytest.raises(
+            ValueError,
+            match=re.escape(
+                "'orientation expected 'v' or 'h'"
+                " (non case sensitive)."
+                " Found l of type <class 'str'>."
+            ),
+        ):
+            current_fixture._plot_summary(
+                current_fixture.daily_route_summary,
+                "route_count_mean",
+                orientation="l",
+            )
+
+    @pytest.mark.filterwarnings("ignore::UserWarning")
+    def test__plot_summary_on_pass(self, gtfs_fixture, tmp_path):
+        """Test plotting a summary when defences are passed."""
+        current_fixture = gtfs_fixture
+        current_fixture.summarise_routes()
+
+        # test returning a html string
+        test_html = gtfs_fixture._plot_summary(
+            gtfs_fixture.daily_route_summary,
+            "route_count_mean",
+            return_html=True,
+        )
+        assert type(test_html) == str, "Failed to return HTML for the plot"
+
+        # test returning a plotly figure
+        test_image = gtfs_fixture._plot_summary(
+            gtfs_fixture.daily_route_summary, "route_count_mean"
+        )
+        assert (
+            type(test_image) == PlotlyFigure
+        ), "Failed to return plotly.graph_objects.Figure type"
+
+        # test saving plots in html and png format
+        gtfs_fixture._plot_summary(
+            gtfs_fixture.daily_route_summary,
+            "route_count_mean",
+            width=1200,
+            height=800,
+            save_html=True,
+            save_image=True,
+            save_pth=pathlib.Path(
+                os.path.join(tmp_path, "save_test", "test_image")
+            ),
+            ylabel="Mean",
+            xlabel="Day",
+            orientation="h",
+            plotly_kwargs={"legend": dict(bgcolor="lightgrey")},
+        )
+
+        # general save test
+        assert os.path.exists(
+            os.path.join(tmp_path, "save_test")
+        ), "'save_test' dir could not be created'"
+        assert os.path.exists(
+            os.path.join(tmp_path, "save_test", "test_image.html")
+        ), "Failed to save summary in HTML"
+        assert os.path.exists(
+            os.path.join(tmp_path, "save_test", "test_image.png")
+        ), "Failed to save summary as a PNG"
+
+        # save test for HTML with an invalid file extension
+        with pytest.warns(
+            UserWarning,
+            match=re.escape(
+                "HTML save requested but accepted type not specified.\n"
```

Comment:
@CBROWN-ONS by accepted type, do you mean the user asked for html as `save_html=True` but wrong file extension provided? If I've interpreted the behaviour correctly, suggest a UserWarning like:
"`save_html` is True but invalid file extension passed to `save_pth`."
Alternatively, you could check the extension matches the correct parameter `save_html` or `save_image` and stomp on the bad file extension with the required one. Good idea to raise a warning in that event, too. Would you rather adjust the warning message or behaviour though?

Reply:
Essentially there isn't really a good answer for this as the user can save both a HTML output and an image at the same time, leading to conflicting file extension options. Because of this it essentially is going to raise a warning each time you choose to save as both.  
A possible adjustment to the error message is to recognise the cases where both `save_image` and `save_html` == `True`. Another option is to generalise the path to only accept paths without file extensions, and then possibly add another param for image file extensions (where there are multiple options).

Reply:
Have you seen the behaviour on lines 627 to 667 in gtfs.validation.py? It essentially ensure that when save_html=True, .html is specified, and the same for save_image (with it's respective accepted formats). This sounds similar to what you're suggesting unless I'm understanding incorrectly.

Reply:
However the output_format suggestion sounds like a good implementation.

Reply:
@r-leyshon Essentially it already checks for the correct file extension, and imputes one if the users file extension is deemed invalid. In theory, the warnings could be converted to print statements as the functionality remains the same. No matter what, the figure will be saved correctly.

```python
 # save the plot if specified (with correct file type)
if save_html:
    if ext.lower() != ".html":
        warnings.warn(
            (
                "HTML save requested but accepted type "
                "not specified.\n"
                "Accepted image formats include ['.html']\n"
                "Saving as .html"
            ),
            UserWarning,
        )
    plotly_io.write_html(
        fig=fig,
        file=os.path.normpath(main + ".html"),
        full_html=False,
    )

if save_image:
    valid_img_formats = [
        ".png",
        ".pdf",
        "jpg",
        "jpeg",
        ".webp",
        ".svg",
    ]
    if ext.lower() not in valid_img_formats:
        warnings.warn(
            (
                "Image save requested but accepted type not "
                "specified.\n"
                f"Accepted image formats include {valid_img_formats}\n"
                "Saving as .png"
            ),
            UserWarning,
        )
        ext = ".png"
    plotly_io.write_image(fig=fig, file=os.path.normpath(main + ext))
```

Reply:
@r-leyshon A PR has now been opened for this refactor. It gives the user less room for error while also raising errors if invalid file types are passed, rather than warnings